wp-content/plugins/automatic-translation/TranslationWpmlHelper.php

```diff
@@ -19,4 +19,48 @@ public function link_wpml($source_post_id, $source_post_type, $translated_post_i
 		}
 		$sitepress->set_element_language_details($translated_post_id, $wpml_post_type, $source_trid, $language_code);
 	}
-}
\ No newline at end of file
+	
+	//not needed yet
+	public function get_translation_post_parent( $post_id, $source_language_code, $target_language_code ) {
+		global $wpdb;
+		
+		
+		//convert single querys to join!
```

Comment:
there is no reason to join manually here. let SQL do that for you, i.e. 

    SELECT post_parent, post_status 
    FROM $wpdb->posts posts
    JOIN {$wpdb->prefix}icl_translations translations_id ON translations_id.element_id = '$post_id'
    JOIN {$wpdb->prefix}icl_translations translations_source  ON translations_source.element_id = posts.ID AND language_code = '$source_language_code' AND translations_source.trid = translations_id.trid

(untested)